kura/examples/org.eclipse.kura.demo.modbus/src/main/java/org/eclipse/kura/demo/modbus/ModbusExample.java

```diff
@@ -44,99 +86,221 @@ public void unsetModbusProtocolDeviceService(ModbusProtocolDeviceService modbusS
 
-        return decimal;
-    }
+    /**
+     * Called after a new set of properties has been configured on the service
+     */
+    private void doUpdate(Map<String, Object> properties) {
+        try {
+
+            for (String s : properties.keySet()) {
+                s_logger.info("Update - " + s + ": " + properties.get(s));
+            }
+
+            // cancel a current worker handle if one if active
+            if (this.m_handle != null) {
+                this.m_handle.cancel(true);
+            }
+
+            if (this.m_protocolDevice != null) {
+                try {
+                    this.m_protocolDevice.disconnect();
+                } catch (ModbusProtocolException e) {
+                    s_logger.error("Failed to disconnect : {}", e.getMessage());
+                }
+            }
+            this.configured = false;
+
+            this.m_properties = properties;
+            modbusProperties = getModbusProperties();
+            if (this.m_properties.get(PUBLISH_TOPIC_PROP_NAME) != null) {
+                this.m_topic = (String) this.m_properties.get(PUBLISH_TOPIC_PROP_NAME);
+            }
+            if (this.m_properties.get(PUBLISH_INTERVAL) != null) {
+                this.m_publish_interval = (Integer) this.m_properties.get(PUBLISH_INTERVAL);
+            }
+
+            if (this.m_properties.get(INPUT_ADDRESS) != null) {
+                inputaddr = (Integer) this.m_properties.get(INPUT_ADDRESS);
+            }
+            else inputaddr=0;
+            if (this.m_properties.get(REGISTER_ADDRESS) != null) {
+                registeraddr = (Integer) this.m_properties.get(REGISTER_ADDRESS);
+            }
+            else registeraddr=0;
 
-    private void doModbusLoop() {
-        while (!this.m_threadShouldStop) {
             if (!this.configured) {
                 try {
                     if (modbusProperties != null) {
                         configureDevice();
-                        // boolean[] ab=m_protocolDevice.readExceptionStatus();
-                        // m_protocolDevice.writeSingleCoil( 2048, false);
-                        // int[] regs=m_protocolDevice.readHoldingRegisters(40000, 8);
-                        // for(int reg:regs)
-                        // s_logger.info(String.valueOf(reg));
-                        // boolean[] ab=m_protocolDevice.readCoils(2048, 8);
-                        // for(boolean b:ab)
-                        // s_logger.info(String.valueOf(b));
-                        initializeLeds();
                     }
                 } catch (ModbusProtocolException e) {
-                    // s_logger.error(e.getMessage());
-                    e.printStackTrace();
+                    s_logger.error("ModbusProtocolException :  {}", e.getMessage());
                 }
-            } else {
+            }
+
+            // schedule a new worker based on the properties of the service
+            int pubrate = 1000;
```

Comment:
And it is already done in the metatype. So you don't need to initialise it to 1000.